common/djangoapps/enrollment/tests/test_api.py

```diff
@@ -8,6 +8,8 @@
 from django.test import TestCase
 from django.test.utils import override_settings
 from django.conf import settings
+
+from course_modes.models import CourseMode
```

Comment:
Is there a test case for when `mode` is passed to `add_enrollment`?  It should check whether the passed in mode overrides the default.